hazelcast/src/main/java/com/hazelcast/internal/partition/operation/AbstractPartitionOperation.java

```diff
@@ -90,8 +101,108 @@
         return operations;
     }
 
-    // must be invoked on partition thread; prepares replication operations within partition thread
-    // does not support differential sync
+    @Nonnull
+    final Collection<ChunkSupplier> collectChunkSuppliers(PartitionReplicationEvent event,
+                                                          Collection<String> serviceNames,
+                                                          ServiceNamespace namespace) {
+        ILogger logger = getLogger();
+        logger.fine("Collecting chunk suppliers...");
+
+        Collection<ChunkSupplier> chunkSuppliers = emptyList();
+
+        NodeEngine nodeEngine = getNodeEngine();
+        for (String serviceName : serviceNames) {
+            Object service = nodeEngine.getService(serviceName);
+            if (!(service instanceof ChunkedMigrationAwareService)) {
+                // skip not chunked services
+                continue;
+            }
```

Comment:
What if we rather assert that service is an instance of `ChunkedMigrationAwareService`? :)

So, from what I understand, with chunked migration we want to replace `Fragmented(MigrationAwareService)` with `Chunked(MigrationAwareService)` where instances of the latter either 'override' replicationOperationPreparation as to make it chunked or default to previous behavior. Thus, since in the upper layer we still have fragmented and non-fragmented service separation, we would want all services that implement `Fragmented` to be instances of `Chunked` as well. Thus, in the future if we have a service that implements  `Fragmented`, but not `Chunked` that service might not get replicated during migration, no?

Reply:
Please see here: https://github.com/hazelcast/hazelcast/pull/20005/files#diff-8ae4f7b6f47cac763ef48111f12148378915f37b7c8b0e5aa266b048336330f8R279

After chunked services are migrated, we also check for `Fragmented` but not `Chunked` ones and migrate them also.

Reply:
Sorry, not sure if the link is broken, but it doesn't show anything. Just goes to `files changed` tab

Reply:
oops! it should be reachable like this: https://github.com/ahmetmircik/hazelcast/blob/354ff1bd9a5957eaf2859e1dc941029da4c00912/hazelcast/src/main/java/com/hazelcast/internal/partition/operation/MigrationRequestOperation.java#L279

Reply:
thanks